ui/shared/components/buttons/SearchResultsLink.jsx

```diff
@@ -48,11 +48,17 @@ const ConnectedSearchResultsLink = connect(null, mapDispatchToProps)(SearchResul
 
 export default ConnectedSearchResultsLink
 
-const INITIAL_GENE_SEARCH = {
+const getGeneSearchProps = af => ({
```

Comment:
What I recommended on the ticket had a very important purpose: pre-generating the dictionary so it will not be re-generated, resulting in unnecessary re-renders of the component. With this change, anytime any prop to the link changes this whole dictionary will be recomputed, and the link will constantly re-render

This also adds an unnecessary and dangerous amount of flexibility. We don't want every gene search link in seqr to be slightly different, we want consistant behavior. In this specific case we need it to be different, but we want that to be a very explicit different case. We don't want the possibility of every possible af in gene search, we want exactly 2 possibilities

If you prefer adding a prop instead of a new component, you can add a boolean `requireRareAf` and then stil precompute the 2 possible initial search constants and toggle between them
```
initialSearch={requireRareAf ? RARE_INITIAL_GENE_SEARCH : INITIAL_GENE_SEARCH}
```

Reply:
Okay, I see. A new rare gene search link component may be more straightforward.